src/spikeinterface/extractors/neoextractors/neuroscope.py

```diff
@@ -43,16 +45,25 @@ class NeuroScopeRecordingExtractor(NeoBaseRecordingExtractor):
     NeoRawIOClass = "NeuroScopeRawIO"
     name = "neuroscope"
 
-    def __init__(self, file_path, stream_id=None, stream_name=None, all_annotations=False):
-        neo_kwargs = self.map_to_neo_kwargs(file_path)
+    def __init__(self, file_path, xml_file_path=None, stream_id=None, stream_name=None, all_annotations=False):
+        neo_kwargs = self.map_to_neo_kwargs(file_path, xml_file_path)
+
         NeoBaseRecordingExtractor.__init__(
             self, stream_id=stream_id, stream_name=stream_name, all_annotations=all_annotations, **neo_kwargs
         )
-        self._kwargs.update(dict(file_path=str(file_path)))
+        self._kwargs.update(dict(file_path=str(file_path), xml_file_path=xml_file_path))
 
     @classmethod
-    def map_to_neo_kwargs(cls, file_path):
-        neo_kwargs = {"filename": str(file_path)}
+    def map_to_neo_kwargs(cls, file_path, xml_file_path=None):
+        # For this because of backwards compatibility we have a strange convention
+        # filename is the xml file
+        # binary_file is the binary file in .dat, .lfp, .eeg
+
+        if xml_file_path is not None:
+            neo_kwargs = {"binary_file": file_path, "filename": xml_file_path}
+        else:
+            neo_kwargs = {"filename": file_path}
```

Comment:
o.O

Reply:
Yeah, the API of neo is showing age here. I tried to document it better on that side:

https://github.com/NeuralEnsemble/python-neo/pull/1290